geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -0,0 +1,268 @@
+   * object, the striped executor will not protect toData. So any methods that modify "members"
+   * needs to be thread safe with toData.
+   */
+
+  @Override
+  public synchronized void toData(DataOutput out, SerializationContext context) throws IOException {
+    super.toData(out, context);
+    InternalDataSerializer.writePrimitiveInt(members.size(), out);
+    for (Map.Entry<byte[], byte[]> entry : members.entrySet()) {
+      byte[] member = entry.getKey();
+      byte[] score = entry.getValue();
+      InternalDataSerializer.writeByteArray(member, out);
+      InternalDataSerializer.writeByteArray(score, out);
+    }
+    InternalDataSerializer.writePrimitiveInt(sizeInBytes, out);
+  }
+
+  @Override
+  public void fromData(DataInput in, DeserializationContext context)
+      throws IOException, ClassNotFoundException {
+    super.fromData(in, context);
+    int size = InternalDataSerializer.readPrimitiveInt(in);
+    members = new Object2ObjectOpenCustomHashMap<>(size, ByteArrays.HASH_STRATEGY);
+    for (int i = 0; i < size; i++) {
+      members.put(InternalDataSerializer.readByteArray(in),
+          InternalDataSerializer.readByteArray(in));
+    }
+    sizeInBytes = InternalDataSerializer.readPrimitiveInt(in);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof RedisSortedSet)) {
+      return false;
+    }
+    if (!super.equals(o)) {
+      return false;
+    }
+    RedisSortedSet redisSortedSet = (RedisSortedSet) o;
+    if (members.size() != redisSortedSet.members.size()) {
+      return false;
+    }
+
+    for (Map.Entry<byte[], byte[]> entry : members.entrySet()) {
+      if (!Arrays.equals(redisSortedSet.members.get(entry.getKey()), (entry.getValue()))) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public int getDSFID() {
+    return REDIS_SORTED_SET_ID;
+  }
+
+  protected synchronized AddOrChange memberAdd(byte[] memberToAdd, byte[] scoreToAdd) {
+    boolean added = (members.put(memberToAdd, scoreToAdd) == null);
+    if (added) {
+      sizeInBytes += calculateSizeOfFieldValuePair(memberToAdd, scoreToAdd);
+    } else {
+      // TODO: calculate size for 'changed'
+    }
+    return added ? AddOrChange.ADDED : AddOrChange.NOOP;
+  }
+
+  private synchronized void membersAddAll(AddsDeltaInfo addsDeltaInfo) {
+    Iterator<byte[]> iterator = addsDeltaInfo.getAdds().iterator();
+    while (iterator.hasNext()) {
+      byte[] member = iterator.next();
+      byte[] score = iterator.next();
+      sizeInBytes += calculateSizeOfFieldValuePair(member, score);
+      members.put(member, score);
+    }
+  }
+
+
+  private synchronized void membersRemoveAll(RemsDeltaInfo remsDeltaInfo) {
+    for (byte[] member : remsDeltaInfo.getRemoves()) {
+      sizeInBytes -= calculateSizeOfFieldValuePair(member, members.get(member));
+      members.remove(member);
```

Comment:
In this case, it is a delta, so we know the member was removed from a different Geode member, and this member receiving the delta should have the same key/values as the member sending the delta (minus the delta of course).  I don't think there is a scenario where you can remove just a score without also removing the key itself....  

Would it ever be possible for the receiving member's backing set to not have the value that was in the Delta-sending member's set?  I guess it would be good to do a `memberRemove()` and check to be sure.  Though since we're not implementing any removal methods yet, this method could potentially be removed and added back along with `membersRemove()` in a PR that adds removal functionality (like `ZREM`).